dashboard/modules/job/job_manager.py

```diff
@@ -489,16 +492,23 @@ async def run(
                             f"failed with exit code {return_code}. No logs available."
                         )
                     await self._job_info_client.put_status(
-                        self._job_id, JobStatus.FAILED, message=message
+                        self._job_id,
+                        JobStatus.FAILED,
+                        message=message,
+                        driver_exit_code=return_code,
                     )
         except Exception:
             logger.error(
                 "Got unexpected exception while trying to execute driver "
                 f"command. {traceback.format_exc()}"
             )
+            return_code = child_process.returncode if child_process else None
```

Comment:
We don't need exit code in this case. It's possible that child_process is not even created.